gulp-tasks/build-packages.js

```diff
@@ -116,13 +116,81 @@ gulp.task('build-packages:clean', gulp.series(
 // This will create one version of the tests for each buildType.
 // i.e. we'll have a browser build for no NODE_ENV and one for 'prod'
 // NODE_ENV and the same for sw and node tests.
-const packageBuilds =constants.BUILD_TYPES.map((buildType) => {
+const packageBuilds = constants.BUILD_TYPES.map((buildType) => {
   return packageRunnner('build-package', buildPackage, buildType);
 });
 
 gulp.task('build-packages:build', gulp.series(packageBuilds));
 
+const convertExportObject = (exportObject, levels = 0) => {
+  let outputStrings = [];
+  const keys = Object.keys(exportObject);
+  keys.forEach((key) => {
+    const value = exportObject[key];
+    if (typeof(value) === 'string') {
+      if (key === value) {
+        outputStrings.push(`${value}`);
+      } else {
+        outputStrings.push(`${key}: ${value}`);
+      }
+    } else {
+      const valueString = convertExportObject(value, levels + 1);
+      outputStrings.push(`${key}: ${valueString}`);
+    }
+  });
+  const padding = '  '.repeat(levels + 1);
+  const closePadding = '  '.repeat(levels);
+  const joinString = `,\n${padding}`;
+  return `{\n${padding}${outputStrings.join(joinString)}\n${closePadding}}`;
+};
+
+const generateBrowserEntry = (pkgPath) => {
```

Comment:
Minor: For `generateBrowserEntry()` and `convertExportObject()`, could we add a short comment summary to remind folks what they're accomplishing? It would be useful to get a thousand foot view when scrolling through without having to parse the relationships between these different pieces.